### PR TITLE
Add full application form

### PR DESCRIPTION
**Pull request by kailanjian**
me and jesse worked on this

### CODE REVIEW

File: backend/api/application_form/migrations/0005_applicationform_birth_date.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    dependencies = [
+        ('application_form', '0004_auto_20200529_0707'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='applicationform',
+            name='birth_date',
+            field=models.DateTimeField(default=datetime.now(), max_length=100),
```
Comment (richyen) [inline backend/api/application_form/migrations/0005_applicationform_birth_date.py:17]: This should be a date field, not a datetime

File: backend/api/application_form/models.py [hunk 2]
```diff
@@ -10,17 +11,38 @@ class Meta:
...
     phone_regex = RegexValidator(regex=r'^\+?1?\d{9,15}$',
                                  message="""
 Phone number must be entered in the format: '+999999999'. Up to 15 digits allowed.
 """)
-    first_name = models.CharField(max_length=200)
-    last_name = models.CharField(max_length=200)
+    first_name = models.CharField(max_length=100)
+    last_name = models.CharField(max_length=100)
+    preferred_name = models.CharField(max_length=100, blank=True)
+    birth_date = models.DateTimeField(max_length=100)
```
Comment (richyen) [inline backend/api/application_form/models.py:29]: date v datetime?

File: backend/api/application_form/tests.py [hunk 3]
```diff
@@ -2,29 +2,48 @@
...
+            major="Science",
+            referral="friend",
+            goals="get better at english"
         )
 
     def test_post_application_form(self):
         self.client.post("/api/v1/applicationForms/", {
             "first_name": "Krikor",
             "last_name": "Ailanjian",
+            "birth_date": "2020-05-07T01:05:00Z",
```
Comment (richyen) [inline backend/api/application_form/tests.py:33]: date v. datetime?

**Review by richyen — CHANGES_REQUESTED** [review]
small design decision -- date v. datetime.  Don't mean to be nitpicky, but just wanted to raise awareness, in case you guys didn't realize there was `date` function also

File: backend/api/application_form/migrations/0005_applicationform_birth_date.py (hunk 1 above)
Comment (kailanjian) [reply backend/api/application_form/migrations/0005_applicationform_birth_date.py:17]: I see, I didn't know there was a date field, I just used the first thing I saw

Comment (richyen) [reply backend/api/application_form/migrations/0005_applicationform_birth_date.py:17]: @douglaschan32167 -- What do you think? Stick with `datetime`, or change to `date`?

**Comment by kailanjian** [comment]
I started changing this to date, however there is an issue which we may want to consider: it looks like JavaScript dates are serialized to the format that DateTime expects rather than Date. For that reason I don't know how to send the date without changing the frontend code substantively. So I can keep looking into this, but not sure if it's worth it

File: backend/api/application_form/models.py (hunk 2 above)
Comment (douglaschan32167) [reply backend/api/application_form/models.py:29]: If it's hard on the frontend, I'm okay with it being a datetime. Still, why do we even need this information in the first place? That's a little bit weird. It also makes the sensitivity of personal information we have much higher. I may need to push back on this.

File: backend/api/application_form/migrations/0005_applicationform_birth_date.py (hunk 1 above)
Comment (douglaschan32167) [reply backend/api/application_form/migrations/0005_applicationform_birth_date.py:17]: I'm okay with sticking with datetime.

File: backend/api/application_form/models.py (hunk 2 above)
Comment (douglaschan32167) [reply backend/api/application_form/models.py:29]: Another option is we override the save function in the model like https://stackoverflow.com/questions/23887961/django-how-can-i-modify-text-before-save-them-to-database/23888157

But I could see how that would make working in the future more confusing. I say we just keep it as a datetime. I posted a question about whether we even need this info in the questions/concerns section of the basecamp.

**Review by douglaschan32167 — APPROVED** [review]

**Pull request merged**